js/popup.js

```diff
@@ -1,486 +1,462 @@
-								value: 		value
-							}
-						];
-	// Add it to a request, payload is what we'll send to QuantiModo
-	var request =		{
-							message: "uploadMeasurements",
-							payload:[
-										{
-											measurements:			measurements,
-											name: 					name,
-											source: 				"QuantiModo",
-											category: 				valueCategory,
-											combinationOperation: 	combineOp,
-											unit:					unit
-										}
-									]
-									
-						};
-	chrome.extension.sendMessage(request, function(responseText) {
-			var response = $.parseJSON(responseText);
-			if(response.success == true)
-			{
-				document.getElementById('addmeasurement-variable-value').value = "";
-				document.getElementById('add-addmeasurement-variable-value').value = "";
-				window.close();
-			}
-			else
-			{
-				alert("Adding a measurement failed.");
-				console.log(responseText);
-			}
-		});
-		
-	clearNotifications();
+var onAddButtonClicked = function () {
+    // Create an array of measurements
+    var name = $("#add-addmeasurement-variable-name").val();
+    var unit = $("#add-addmeasurement-variable-unit").val();
+    var value = $("#add-addmeasurement-variable-value").val();
+    var valueCategory = $("#addmeasurement-variable-category").val();
+    var combineOp = $("#combineOperation").val();
+    var datetimeString = $("#add-addmeasurement-variable-date").val();
+
+    //alert (Date.parse(datetimeString)) ;
+    datetimeString = datetimeString.replace("AM", "");
+    datetimeString = datetimeString.replace("PM", "");
+    //
+
+
+    //year, month, day, hours, minutes, seconds, milliseconds
+    var hour = $('#add-addmeasurement-variable-timeh').val();
+    var min = $('#add-addmeasurement-variable-timem').val();
+    var ap = $('#add-addmeasurement-variable-timeap').val();
+    var datetime = new Date(datetimeString);
+
+    /*
+     datetime.setHours(parseInt(hour) + (ap * 12));
+     datetime.setMinutes(min);
+     datetime.setSeconds(0);
+     */
+
+    if (name == '') {
+        alert("Please enter the variable name.");
+        return;
+    }
+    if (value == '') {
+        alert("Please enter the value.");
+        return;
+    }
+    //var variable = getVariableWithName(name);
+    var measurements = [
+        {
+            timestamp: Math.floor(datetime.getTime() / 1000),
+            value: value
+        }
+    ];
+    // Add it to a request, payload is what we'll send to QuantiModo
+    var request = {
+        message: "uploadMeasurements",
+        payload: [
+            {
+                measurements: measurements,
+                name: name,
+                source: "QuantiModo",
+                category: valueCategory,
+                combinationOperation: combineOp,
+                unit: unit
+            }
+        ]
+
+    };
+    chrome.extension.sendMessage(request, function (responseText) {
+        var response = $.parseJSON(responseText);
+        if (response.success == true) {

```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/m_3/QuantiModo-Chrome-Extension/file/832637814/issues/source?bid=2779649&fileBranchId=2790922#l293)